Digipost/APIClient+Document.swift

```diff
@@ -21,28 +19,20 @@ extension APIClient {
                 return [Constants.APIClient.AttributeKey.location : Constants.FolderName.folder, Constants.APIClient.AttributeKey.subject : name, Constants.APIClient.AttributeKey.folderId: documentFolder.folderId.stringValue]
             }
             }()
-        let task = urlSessionTask(httpMethod.post, url: document.updateUri, parameters: parameters, success: success) { (error) -> () in
-            if (error.code == Constants.Error.Code.oAuthUnathorized ) {
-                self.changeName(document, newName: name, success: success, failure: failure)
-            } else {
-                failure(error: error)
-            }
+
+        validateFullScope {
+            let task = self.urlSessionTask(httpMethod.post, url: document.updateUri, parameters: parameters, success: success,failure: failure)
+            task?.resume()
```

Comment:
`task` should probably never be Nil anyways, so we could just force unwrap and let it :fire: if it for some incomprehensible reason is. Then we at least will know about it, rather than failing silently.